dbt/include/synapse/macros/adapters.sql

```diff
@@ -124,26 +124,36 @@
 {%- endmacro %}
 
 {% macro synapse__get_columns_in_relation(relation) -%}
-  {% call statement('get_columns_in_relation', fetch_result=True) %}
-      SELECT
-          column_name,
-          data_type,
-          character_maximum_length,
-          numeric_precision,
-          numeric_scale
-      FROM
-          (select
-              ordinal_position,
-              column_name,
-              data_type,
-              character_maximum_length,
-              numeric_precision,
-              numeric_scale
-          from INFORMATION_SCHEMA.COLUMNS
-          where table_name = '{{ relation.identifier }}'
-            and table_schema = '{{ relation.schema }}') cols
-
+  {# hack because tempdb has no infoschema see: #}
+  {# https://stackoverflow.com/questions/63800841/get-column-names-of-temp-table-in-azure-synapse-dw #}
+  {% if relation.identifier.startswith("#") %}
+    {% set tmp_tbl_hack = relation.incorporate(
+      path={"identifier": relation.identifier.replace("#", "") ~ '_tmp_tbl_hack'},
+      type='table')-%}
+
+    {% do  drop_relation(tmp_tbl_hack) %}
```

Comment:
The only concern here is if the temp table shares a name with another real model. I see the temp suffix used in the snapshot is `pseudotmp`:
https://github.com/dbt-msft/dbt-synapse/blob/b6d2015aa72ef0c3302ea193c0ea09417a88efa5/dbt/include/synapse/macros/materializations/snapshot/snapshot.sql#L7

So if there's a snapshot called `whatever`, and another real object called `whatever_pseudotmp`, we'd incidentally drop the latter. I think that's unlikely :) And users can override that behavior if they really need to, by reimplementing `build_snapshot_staging_table` and/or `make_temp_relation`.

Reply:
good catch. actually that line was the previous workaround that @NandanHegde15 and I came up with, then promptly forgot about. We're talking it pseudo temp because it was a real table.

But now that we have a "proper" hack, we're all aboard the global project train to Snapshotville.